wfe-core/src/main/java/ru/runa/wfe/execution/ExecutionContext.java

```diff
@@ -420,4 +423,22 @@ private void updateRelatedObjectsDueToDateVariableChange(String variableName) {
             log.info(String.format("Changed dueDate for %s from %s to %s", job, oldDate, job.getDueDate()));
         }
     }
+
+    private void updateMessageSelectorIfExists(ProcessDefinition processDefinition, Variable<?> variable, //
+            Process process, Object oldValue) {
+        List<Token> tokenList = tokenDao.findByProcessAndMessageSelectorLikeAndExecutionStatusIsNotEnded(process,
+                Utils.MESSAGE_SELECTOR_VALUE_DELIMITER + TypeConversionUtil.convertTo(String.class, oldValue));
+        for (Token token : tokenList) {
+            Node node = processDefinition.getNodeNotNull(token.getNodeId());
+            if (!(node instanceof BaseMessageNode)) {
+                continue;
```

Comment:
throw exception in this case; message selector can be defined only for catch event message node which waits a signal